pkg/operator/podsecurityreadinesscontroller/labelsyncer.go

```diff
@@ -0,0 +1,108 @@
+package podsecurityreadinesscontroller
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	psapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	syncerControllerName  = "pod-security-admission-label-synchronization-controller"
+	policyControllerName  = "cluster-policy-controller"
+	labelSyncControlLabel = "security.openshift.io/scc.podSecurityLabelSync"
+)
+
+var (
+	alertLabels = []string{
+		psapi.WarnLevelLabel,
+		psapi.AuditLevelLabel,
+	}
+)
+
+func isNSControlled(ns *corev1.Namespace) bool {
```

Comment:
What is this actually supposed to do? The `HasAll` in return is making this quite confusing.

Why do you care about the warn label?
Why do you ignore the enforce label?

Reply:
We are ignoring use cases, where enforce label is set as we don't do that. We assume that the customer took ownership of it.
We want to ensure that we are responsible for audit and warn labels, if the `labelSyncControlLabel` is not set.

Reply:
> We are ignoring use cases, where enforce label is set as we don't do that.

right, but if user sets the enforced label, you don't need to worry about that namespace triggering violations, right?
> Those labels are being used to search for potential upcoming violations

Used by whom? Why do we care if a user gets a warning when they try to create a pod?

Reply:
Discussion: We should only consider labels we set and most probably only audit.